clesperanto/core/plugin_function.py

```diff
@@ -0,0 +1,105 @@
+        per default, we create output images of the same shape as input images
+    :return: return value of the executed function
+    """
+
+
+    @wraps(function)
+    def worker_function(*args, **kwargs):
+        # determine argument spec and default values, values are given as args
+        argument_specification = inspect.getfullargspec(function)
+        defaults_values = argument_specification.defaults
+
+        target_arguments = [None] * len(argument_specification.args)
+
+        arg_counter = 0
+        default_counter = 0
+        any_input = None
+        for argument in argument_specification.args:
+            #print("---\nparsing argument " + argument)
+            if (len(args) > arg_counter):
+                value = args[arg_counter]
+                #print("value")
+                #print(value)
+            else:
+                value = None
+
+            # was the argument annotated?
+            type_annotation = argument_specification.annotations.get(argument);
+            if (type_annotation is not None):
+                if (type_annotation is OCLArray):
+                    # annotated as OpenCL array
+                    if (isinstance(value, OCLArray)):
+                        # value is also OpenCL, pass it
+                        target_arguments[arg_counter] = value
+                        any_input = value
+                    else:
+                        # value is not OpenCL
+                        if (value is not None):
+                            # convert the value to OpenCL. push is magic and can convert anything
+                            value = push(value)
+                            any_input = value
+                            target_arguments[arg_counter] = value
+                        else:
+                            # create a new output image with specified/default creator
+                            target_arguments[arg_counter] = output_creator(any_input)
+
+                else:
+                    # any other type
+                    target_arguments[arg_counter] = value
+            else:
+                # it wasn't annotated
+                if (value is not None):
+                    # if it's something, hand it over
+                    target_arguments[arg_counter] = value
+                else:
+                    # if it's not set, hand over default values
+                    if (len(defaults_values) > default_counter):
+                        target_arguments[arg_counter] = defaults_values[default_counter]
+                        default_counter += 1
+                    else:
+                        target_arguments[arg_counter] = None
+
+            arg_counter += 1
+
+        #print("Got arguments")
+        #print(args)
+        #print("Will pass arguments")
+        #print(target_arguments)
+
+        # execute function with determined arguments
+        return function(*target_arguments)
```

Comment:
To match my assumption of some arguments being keyword only, `target_arguments` should be a dictionary and then this would be 
```suggestion
        return function(**target_arguments)
```

Reply:
Great! I should have seen that target_arguments should be a dictionary :-)